src/Type/Errors/Pretty.hs

```diff
@@ -114,3 +117,34 @@ type family ToErrorMessage (t :: k) :: ErrorMessage where
     ToErrorMessage (t :: Symbol) = 'Text t
     ToErrorMessage (t :: ErrorMessage) = t
     ToErrorMessage t = 'ShowType t
+
+-- | Internal type family to go from a `Type` to a `Symbol` representation.
+-- The names for types without `Generic` instances have to be defined
+-- manually, and we do so for a handful of common types.
+type family TypeName (a :: Type) :: Symbol where
+  TypeName Char = "Char"
+  TypeName String = "String"
+  TypeName Int  = "Int"
+  TypeName (Maybe a) = AppendSymbol "Maybe " (TypeName a)
+  TypeName [a] = AppendSymbol "[" (AppendSymbol (TypeName a) "]")
+  TypeName (D1 ('MetaData name _ _ _) _ _) = name
+  TypeName a = TypeName (Rep a ())
```

Comment:
I think this approach is not extensible. Users of `type-errors-pretty` cannot create new instances of `TypeName`. And the library cannot provide instances for all types 😞 

To solve this problem nicely, the `CommaSeparated` type family can return a type of kind `ErrorMessage` instead of `Symbol`. The type-level operator `<>` already converts types and type-level symbols to a nice stringified representation that can be later reused in error messages.